### ISSUE TITLE
view_photo nn gives: FAIL: 92: RPC_CALL_FAIL 400: FILEREF_UPGRADE_NEEDED

### ISSUE THREAD

**Issue by gurucubano** [vysheng/tg#1629]
Since today any download of an image either gets stuck or gives the message:

 ```FAIL: 92: RPC_CALL_FAIL 400: FILEREF_UPGRADE_NEEDED```

Any hints?

**Comment by gurucubano** [comment]
With some more tries it gives as the error message:

```
> view_photo 34
FAIL: 92: RPC_CALL_FAIL 400: FILEREF_UPGRADE_NEEDED
 *** 1572204650.368052 Notification API_ERROR_UPDATE_UNOFFICIAL: Error. Please update your app to the latest version or switch to the official version of Telegram.
[0%Down]>
```
Maybe the fix is just change some string of the used protocol version?

**Comment by Flohack74** [comment]
No, this means you really need to upgrade. Telegram in general does not just do cosmetic changes, if there is a reason to block old versions then its substantial. I can recommend every developer to switch their Telegram implementations to https://github.com/tdlib/td - This is much easier to keep up with all changes than to maintain all MTproto API levles themselves.

**Comment by gurucubano** [comment]
Hmm, magically it started working again, without having changed anything in the tg cmd client. And it works also again in the UBports tg client 2.5.5.0. Maybe telegram.org got so many complaints, that they rolled back some change. Don't know ...

**Comment by radishboo** [comment]
@gurucubano 
hello there, how is the thing going on ?
I just got the same problem, and don't know why.

**Comment by psychosys** [comment]
> 
> 
> Hmm, magically it started working again, without having changed anything in the tg cmd client. And it works also again in the UBports tg client 2.5.5.0. Maybe telegram.org got so many complaints, that they rolled back some change. Don't know ...

this project is dead ..

**Comment by PedroLopes** [comment]
I am getting a very simular error too with load_document 

`Notification API_ERROR_UPDATE_UNOFFICIAL: Error. Please update your app to the latest version or switch to the official version of Telegram.`

and.. load_photo unfortunately never works (but no error) which is suspicious too. Anybody has a clue?

**Comment by gurucubano** [comment]
Magically, sometime it refuses to download images, sometimes it works (like at the moment for example). 

The problem is that there is no developer taking care of this really nice piece of software (a UNIX cmd line client for tg). The UBports project (ubports.com) developed a new tg app named TELEports for the UBports mobile, but nobody cares about this cmd line client. I'm not familiar with the tg API and so do not know how much effort it would be to switch to this. In any case it's a pity.

**Comment by Paulchen232** [comment]
Perhaps the fork [kenorb-contrib/tg](https://github.com/kenorb-contrib/tg/blob/master/README.md) works better. Installing it with `git clone --recursive https://github.com/kenorb-contrib/tg.git && cd tg`

**Comment by Flohack74** [comment]
I can just repeat what I said above: Telegram is in the process of deprecating their old APIs, and all 3rd party clients are urged to switch to the new tdlib implementation which will use on of the newest API revisions. Over the years Telegram published nearly 100 API revisions, and we need to understand that by that number of versions breaking changes cannot be avoided, and one cannot expect that very old versions are still being maintained.
If you get a message that your client is too old then its for a reason. Its then really too old...

**Comment by gurucubano** [comment]
El día domingo, enero 12, 2020 a las 02:04:06a. m. -0800, Florian Leeber escribió:

> I can just repeat what I said above: Telegram is in the process of deprecating their old APIs, and all 3rd party clients are urged to switch to the new tdlib implementation which will use on of the newest API revisions. Over the years Telegram published nearly 100 API revisions, and we need to understand that by that number of versions breaking changes cannot be avoided, and one cannot expect that very old versions are still being maintained.
> If you get a message that your client is too old then its for a reason. Its then really too old...
> 

Florian, how can I check if the new fork of telegram-cli below:

git clone --recursive https://github.com/kenorb-contrib/tg.git

uses the new API?

	matthias

**Comment by Flohack74** [comment]
It acutally uses this telegram lib: https://github.com/kenorb-contrib/tgl - which is in my view still the old way of accessing Telegram, by writing the client yourself. tdlib is kinda abstraction library that is maintained by Telegram, and the developer does not have to worry about details of the protocol, including the sensitive encryption stuff. Specifically https://github.com/kenorb-contrib/tgl/blob/57f1bc41ae13297e6c3e23ac465fd45ec6659f50/tgl.h#L52 suggests that this client is at API level 45, which is very old already. You might want to try to motivate the maintainer of the fork to refactor the App to change to tdlib.
